alkos/kernel/include/init.hpp

```diff
@@ -3,4 +3,9 @@
 
 void KernelInit();
 
+/**
+ * @brief must be defined by arch code to set up all architecture dependent interrupt settings
+ */
+void KernelArchInit();
```

Comment:
This is something to talk about.
I believe we can simplify the code by having just 2 setup functions for the arch
one being something like PreKernelInit32, other being like PreKernelInit64.
A PreKernelInit64 would setup things like IDT along with other things that we already set up. This way we would lessen the amount of functions that are required to be provided by the Arch as an ABI. Which translates to less places where we can fuck something up.
TLDR
I think this should be in arch during boot sequence and not here.
unless for some reason architecture specific things need to be setup after boot32, boot64 AND kernel

Reply:
move do arch code